### PR TITLE
CLIENT-2014

### PR DESCRIPTION
**Pull request by DomPeliniAerospike**
- [CLIENT-2000] - Added ARM architecture support for all currently supported OS systems
- Added support for:
  - Node.js 19
  - Ubuntu 22.04
  - RHEL 9 
- Dropped support for:
  - Node.js 10, 12, 17
  - RHEL 7
  - Debian 8, 9

[CLIENT-2000]: https://aerospike.atlassian.net/browse/CLIENT-2000?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -87,12 +82,10 @@ Aerospike.connect(config)
...
-The client is compatible with Node.js 10, 12 (LTS), 14 (LTS), 16 (LTS) and 18 (LTS). It
-supports the following operating systems: CentOS/RHEL 7/8, Debian 8/9/10/11,
-Ubuntu 18.04/20.04, as well as many Linux distributions compatible with one of
-these OS releases. macOS is also supported. The client port to Windows is a
-community supported project and suitable for application prototyping and
-development.
+The client is compatible with Node.js 19, 18 (LTS), 16 (LTS) and 14 (LTS). It
+supports the following operating systems: RHEL 7/8/9, Debian 10/11,
+Ubuntu 18.04/20.04/22.04(Focal, Jammy, Bionic), as well as many Linux distributions compatible with one of
+these OS releases. macOS versions 11/12/13 are also supported.
```
Comment (juliannguyen4) [inline README.md:88]:
```suggestion
The client is compatible with Node.js 19, 18 (LTS), 16 (LTS) and 14 (LTS).
It supports the following operating systems:
- RHEL 7/8/9
- Debian 10/11
- Ubuntu 18.04/20.04/22.04 (Focal, Jammy, Bionic)
- Many Linux distributions compatible with one of the above OS releases.
- macOS versions 11/12/13 are also supported.
```
Formatting these into a list makes it easier to read

**Review by juliannguyen4 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (justinlee-aerospike) [reply README.md:88]: @DomPeliniAerospike I thought that macOS is not compatible with with Node.js 14?  If so you need to mention that here.

**Review by justinlee-aerospike — DISMISSED** [review]
Minor requested change, make sure to mention all the incompatibilities around MacOs and Node.js 14.  Otherwise, changes look good.

**Comment by justinlee-aerospike** [comment]
One more thing - can we get the checks to run clean?  Looks like we have a lot of failures.

File: scripts/build-package.ps1 [hunk 2]
```diff
@@ -33,11 +33,9 @@ function build_nodejs_client {
...
-build_nodejs_client v16.14.0
-build_nodejs_client v17.8.0
-build_nodejs_client v18.0.0
+build_nodejs_client v14.21.1
+build_nodejs_client v16.18.1
+build_nodejs_client v18.12.1
+# build_nodejs_client v19.1.0
 
-nvm use v16.14.0
+nvm use v18.12.1
```
Comment (realmgic) [inline scripts/build-package.ps1:41]: I'm pretty sure we moved this entire flow to use github workflows a while back and not use the script anymore

Comment (DomPeliniAerospike) [reply scripts/build-package.ps1:41]: I will need to review the windows workflow, Ramaraj thought I should use this script, but it doesn't work as intended, and this comment may explain why.  I will review this workflow in a future iteration.

File: README.md [hunk 3]
```diff
@@ -223,13 +222,13 @@ by requiring the module by name.
 To install `aerospike` as a dependency of your project, in your project directory run:
 
 ```bash
-$ npm install aerospike
+npm install aerospike
 ```
```
Comment (artanderson) [inline README.md:226]: Is npm the recommended installation? Do you need to go through all the prerequisites first, or should most folks be able to run `npm install aerospike`? 

Keeping in line with our competitors and most other npm packages, we may want to move installation to the beginning of the readme and make the installation process more clearly defined. If `npm install aerospike` is all you need in most situations, we should highlight that.

File: scripts/build-package.ps1 (hunk 2 above)
Comment (DomPeliniAerospike) [reply scripts/build-package.ps1:41]: By linux flow do you mean scripts/build-package.sh? This is used in Github workflows, but I also use this to create binaries for the NPM package.

**Comment by juliannguyen4** [comment]
Looks good to me

**Review by juliannguyen4 — DISMISSED** [review]

**Review by artanderson — DISMISSED** [review]
LGTM

**Review by mcoberly2 — APPROVED** [review]
LGTM

**Comment by mcoberly2** [comment]
Justin Lee's approval via slack: 
image (5)

**Pull request merged**